src/controller/CHIPDeviceController.cpp

```diff
@@ -70,6 +70,7 @@ using namespace chip::Encoding;
 constexpr const char kPairedDeviceListKeyPrefix[] = "ListPairedDevices";
 constexpr const char kPairedDeviceKeyPrefix[]     = "PairedDevice";
 constexpr const char kNextAvailableKeyID[]        = "StartKeyID";
+constexpr const uint32_t kPairingTimeout          = 30 * kMillisecondPerSecond;
```

Comment:
Why does the pairing hang in your case? Will this be needed after switching to CRMP transport which AFAIK supports timeouts for individual messages? My concern is that 30s may be too little for the scenario where a user must enter network credentials by hand (e.g. in the Android app). Maybe at least the timer could be cancelled at the network provisioning phase?

Reply:
Moved it to OnRendezvousStatusUpdate.